File: 02. CLIMOV/CLIMOV_JAVA/app/src/main/java/ec/edu/monster/LoginActivity.java
package ec.edu.monster;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import ec.edu.monster.controlador.AuthControlador;
import ec.edu.monster.modelo.AuthModelo;
import ec.edu.monster.vista.AuthVista;

public class LoginActivity extends AppCompatActivity implements AuthVista {
    private EditText etUsuario, etPassword;
    private TextView tvError;
    private AuthControlador authControlador;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        etUsuario = findViewById(R.id.etUsuario);
        etPassword = findViewById(R.id.etPassword);
        tvError = findViewById(R.id.tvError);
        Button btnLogin = findViewById(R.id.btnLogin);

        AuthModelo authModel = new AuthModelo();
        authControlador = new AuthControlador(authModel, this, this);

        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String usuario = etUsuario.getText().toString().trim();
                String password = etPassword.getText().toString().trim();
                authControlador.validateCredentials(usuario, password);
            }
        });
    }

    @Override
    public void showError(String message) {
        tvError.setText(message);
        tvError.setVisibility(View.VISIBLE);
    }

    @Override
    public void navigateToMain() {
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }
}

File: 04. REST_DONTNET_CONUNI_GRO04/Nueva carpeta/CLIMOV_DOTNET/app/src/main/java/ec/edu/monster/views/LoginActivity.kt
package ec.edu.monster.views

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import ec.edu.monster.controller.LoginController
import ec.edu.monster.model.UserModel
import kotlinx.coroutines.launch
import ec.edu.monster.databinding.ActivityLoginBinding

class LoginActivity : AppCompatActivity() {
    private lateinit var binding: ActivityLoginBinding
    private val loginController = LoginController()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Valores por defecto para pruebas
        binding.etUsername.setText("MONSTER")
        binding.etPassword.setText("MONSTER9")

        binding.btnLogin.setOnClickListener {
            val username = binding.etUsername.text.toString()
            val password = binding.etPassword.text.toString()

            if (username.isNotEmpty() && password.isNotEmpty()) {
                login(UserModel(username, password))
            } else {
                binding.tvError.text = "Por favor complete todos los campos"
            }
        }
    }

    private fun login(user: UserModel) {
        binding.progressBar.visibility = View.VISIBLE
        binding.tvError.text = ""

        lifecycleScope.launch {
            val result = loginController.login(user)

            binding.progressBar.visibility = View.GONE

            if (result.success) {
                Toast.makeText(
                    this@LoginActivity,
                    "Login exitoso",
                    Toast.LENGTH_SHORT
                ).show()

                startActivity(
                    Intent(this@LoginActivity, ConversionActivity::class.java)
                )
                finish()
            } else {
                binding.tvError.text = result.error
            }
        }
    }
}